keras/src/layers/layer.py

```diff
@@ -1602,6 +1602,13 @@ def rematerialized_call(self, layer_call, *args, **kwargs):
             Rematerialized layer's `call` method.
         """
 
+        def remated_function():
+            get_function_only = kwargs.pop("get_function_only", False)
```

Comment:
Hmm... I'm not a big fan of this secret argument.

Instead, let's make `rematerialized_call` always return the rematted function only and not the result (which I believe is the way you had it initially). You'll simply need to call it afterwards.

The confusing part about it is that you still need to pass the arguments, purely to compute the size in the "larger_than" mode.

So it's going to look like this:
```python
self.rematerialized_call(self.call, *args, **kwargs)(*args, **kwargs)
```
Which is weird, but I think it's more important to be consistent.

Reply:
Done!